### DISCUSSION TITLE
Require and install dependency

### DISCUSSION THREAD

**Discussion by mariuskoeppen** [cargo-generate/cargo-generate#1316] [Q&A]
Is there a way for me to require a cargo (or other) dependency to be installed? Something like [the requirement spec for cargo-generate](https://cargo-generate.github.io/cargo-generate/templates/require_version.html) but for other programs. I think about something like that in cargo-generate.toml 
```toml
[require]
trunk = ">=0.20.0"
```
If the dependency is not met, prompt the user to install it. Is there already a way for me to achieve this?

**Comment by sassman** [comment]
Right now this is not possible.
But I would like to understand the use case a bit better, can you elaborate where you would like to use in this global installed tool in the whole process of the template generation?

But in general introducing such a feature that would check for global installed tools might introduce a complexity a complexity.

**Reply by mariuskoeppen** [reply]
This would be useful in a case, where the generated template is supposed to be developed with the required tool. Lets look at this starter template for [leptos + trunk](https://github.com/leptos-rs/start-trunk). After using `cargo generate` to generate the template, the user needs to run `trunk serve` for development. But what if `trunk` is not installed? 
It would be a nice quality of life feature to check in `cargo generate` if the `trunk` command is installed, and if not, prompt the user "Trunk is not installed. Would you like to install it?" or at least some hint, like "This template requires trunk, run `cargo install trunk` to install."
Further you could check for minimum version requirement, so something like "You need at least version x.y.z of this tool. Do you want to update?" etc.

**Reply by sassman** [reply]
I see, but why would the generated project then not define trunk as a dependency itself eg as a build dependency?

At least I would do this with other tools when my project depends on it.
